### PR TITLE
Added Binary_Tree_Top_View 

### PR DESCRIPTION
**Pull request by helios1101**
In reference to pr #509 
Contains cpp, py file.
- [ ] Binary_Tree_TopView

### CODE REVIEW

File: Binary_Tree_Top_View/Binary_Tree_Top_View.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+hd = 0
+root.hd = hd
+
+# push node and horizontal
+# distance to queue
+q.append(root)
+
+while(len(q)) :
+root = q[0]
+hd = root.hd
```
Comment (jainaman224) [inline Binary_Tree_Top_View/Binary_Tree_Top_View.py]: Please fix variable name.

File: Binary_Tree_Top_View/Binary_Tree_Top_View.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+# count function returns 1 if the
+# container contains an element
+# whose key is equivalent to hd,
+# or returns zero otherwise.
+if hd not in m:
+m[hd] = root.data
```
Comment (jainaman224) [inline Binary_Tree_Top_View/Binary_Tree_Top_View.py]: Indentation issues.

File: Binary_Tree_Top_View/Binary_Tree_Top_View.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+self.hd = 0
+
+# function should print the topView
+# of the binary tree
+def topview(root) :
+
+if(root == None) :
+return
+q = []
+m = dict()
```
Comment (jainaman224) [inline Binary_Tree_Top_View/Binary_Tree_Top_View.py]: Use better variable names.

File: Binary_Tree_Top_View/Binary_Tree_Top_View.py (hunk 1 above)
Comment (helios1101) [reply Binary_Tree_Top_View/Binary_Tree_Top_View.py]: Done

File: Binary_Tree_Top_View/Binary_Tree_Top_View.py (hunk 3 above)
Comment (helios1101) [reply Binary_Tree_Top_View/Binary_Tree_Top_View.py]: q is queue 
m is mp now to show map.

**Pull request merged**